pkg/dwarf/op/regs.go

```diff
@@ -72,12 +112,25 @@ func (regs *DwarfRegisters) BP() uint64 {
 
 // AddReg adds register idx to regs.
 func (regs *DwarfRegisters) AddReg(idx uint64, reg *DwarfRegister) {
-	if idx >= uint64(len(regs.Regs)) {
+	if idx >= uint64(len(regs.regs)) {
 		newRegs := make([]*DwarfRegister, idx+1)
-		copy(newRegs, regs.Regs)
-		regs.Regs = newRegs
+		copy(newRegs, regs.regs)
+		regs.regs = newRegs
+	}
+	regs.regs[idx] = reg
+}
+
+// ClearRegisters clears all registers except for PC, SP and BP
+func (regs *DwarfRegisters) ClearRegisters() {
```

Comment:
I think the name should reflect that this is selective in which registers it clears (e.g. `ClearGeneralPurposeRegisters` or something). Also, why be selective?